src/frontend/src/icp/validation/ic-token.validation.ts

```diff
@@ -29,3 +30,8 @@ export const isIcCkToken = (token: Token): token is IcCkToken => {
 
 export const isNotIcCkToken = (token: Token): token is Exclude<Token, IcCkToken> =>
 	!isIcCkToken(token);
+
+export const hasIndexCanister = (token: IcToken): token is IcToken & IcCanistersStrict =>
+	nonNullish(token) && isIcTokenCanistersStrict(token);
+
+export const hasNoIndexCanister = (token: IcToken): token is IcToken => !hasIndexCanister(token);
```

Comment:
If a token has an index canister, it's also an `IcToken`. Maybe better to just type the return as a `boolean` and do not specify assertion here?

Reply:
@peterpeterparker hmm, yes might be the easiest.